server/balancer_test.go

```diff
@@ -247,31 +247,86 @@ func (s *testStorageBalancerSuite) TestReplicaScore(c *C) {
 	c.Assert(sb.cache.get(1), IsTrue)
 
 	// Store 4 has smaller storage ratio than store 2.
-	tc.addLabelsStore(4, 1, 0.1, map[string]string{"zone": "z1", "rack": "r2", "host": "h3"})
+	tc.addLabelsStore(4, 1, 0.1, map[string]string{"zone": "z1", "rack": "r2", "host": "h1"})
 	checkTransferPeer(c, sb.Schedule(cluster), 2, 4)
 
 	// Store 5 has smaller storage ratio than store 1.
 	tc.addLabelsStore(5, 1, 0.2, map[string]string{"zone": "z1", "rack": "r1", "host": "h1"})
 	sb.cache.delete(1) // Delete store 1 from cache, or it will be skipped.
 	checkTransferPeer(c, sb.Schedule(cluster), 1, 5)
 
-	// Store 6 has smaller storage ratio than store 5 and different rack with other stores.
-	tc.addLabelsStore(6, 1, 0.1, map[string]string{"zone": "z1", "rack": "r1", "host": "h2"})
+	// Store 6 has smaller storage ratio than store 5.
+	tc.addLabelsStore(6, 1, 0.1, map[string]string{"zone": "z1", "rack": "r1", "host": "h1"})
 	checkTransferPeer(c, sb.Schedule(cluster), 1, 6)
 
-	// Take down 4,5,6
+	// Store 7 has the same storage ratio with store 6, but in a different host.
+	tc.addLabelsStore(7, 1, 0.2, map[string]string{"zone": "z1", "rack": "r1", "host": "h2"})
+	checkTransferPeer(c, sb.Schedule(cluster), 1, 7)
+
+	// If store 7 is not available, we wait.
+	tc.setStoreDown(7)
+	c.Assert(sb.Schedule(cluster), IsNil)
+	c.Assert(sb.cache.get(1), IsTrue)
```

Comment:
Store 1 is not able to balance temporarily when 7 is down, we can skip it for a while.